zerver/views/__init__.py

```diff
@@ -837,6 +830,11 @@ def home(request):
     request._email = request.user.email
     request.client = get_client("website")
 
+    # If a user hasn't signed the current Terms of Service, send them there
+    if settings.TERMS_OF_SERVICE is not None and \
+       int(settings.TOS_VERSION.split('.')[0]) > user_profile.major_tos_version():
+        return accounts_accept_terms(request)
```

Comment:
accounts_accept_terms will do either a render_to_response or redirect depending on whether we need to show the TOS form or not. (Not sure if that answers the question.)